### DISCUSSION TITLE
Meta does not work on server side

### DISCUSSION THREAD

**Discussion by mirkofisic** [quasarframework/quasar#11266] [CLI - SSR mode] [answered]
I want to use Meta plugin on server side to chnage head title and description during render, but on server env I got error:

> [Vue warn]: Unhandled error during execution of setup function 
  at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > >
[Quasar Dev Webserver] / -> error during render
TypeError: Cannot read properties of undefined (reading 'push')
    at useMeta (C:\projects\quasar\testpr\node_modules\quasar\dist\quasar.cjs.prod.js:6:465788)
    at setup (chunk-src_pages_Index_vue.js:76:54)

Also I have to note that I use SSR and PWA

```
<script async setup>
import { useMeta } from 'quasar'
const metaData = {
  title: 'Test Index',
  meta: {
    description: { name: 'description', content: 'Page 1' },
    keywords: { name: 'keywords', content: 'Quasar website' }
  }
}
if (process.env.SERVER) {
  useMeta(metaData) // does not work
}
console.log('IsServer:', process.env.SERVER)
</script>
```
I use Meta inside page.
Does the Meta only work on the client side? How can I then change the title and description, it is important to me because of SEO?

**Comment by rstoenescu** [comment]
I think the problem is with the `<script setup>` support. Will look into it -- but please open up a ticket for it.
Also, don't guard the useMeta() call by process.env.SERVER --- this needs to run on both server-side and on subsequent navigation in client-side also.

**Reply by mirkofisic** [reply]
I will describe more detail, I tried to use the meta inside <script setup> I also tried to use inside the setup () function when defining the component the error is always the same, it looks like that meta only works with Options API and createMetaMixin hook, but I dont want to use Options API, whole application is Composition API, please can you review this its very important to us at this moment!

```
export default {
  setup () {
    // needs to be called in setup()
    useMeta(metaData) // the same error!
  }
}
```

**Comment by rstoenescu** [comment, marked as answer]
You have to first install the Meta plugin.

<img width="454" alt="image" src="https://user-images.githubusercontent.com/4665641/141108834-f3fb968e-6f44-4bb5-a8f3-811816ec5e5c.png">

**Reply by mirkofisic** [reply]
I was sure that I installed it, sorry i make mistake. thank you very much.
